src/actions.js

```diff
@@ -87,8 +87,44 @@ export function fetchOrganisations() {
       .then(responseObj => responseObj.json())
       .then(responseData => {
         const data = responseData.results;
+        const allOrganisations = data.map(organisation => {
+          //use organisation uuid without dashes only
+          return {
+            ...organisation,
+            uuid: organisation.uuid.replace(/-/g, "")
+          };
+        })
+        const availableOrganisations = allOrganisations.filter(e => {
+          return (
+            e.roles.find(e => e === "manager") ||
+            e.roles.find(e => e === "admin") ||
+            e.roles.find(e => e === "supplier") 
+            // users with only role "user" are no longer allowed in the management page. 
+            // ||
+            // e.roles.find(e => e === "user")
+          );
+        })
+
+        dispatch({ type: RECEIVE_ORGANISATIONS, all:allOrganisations, available: availableOrganisations});
 
-        dispatch({ type: RECEIVE_ORGANISATIONS, data });
+        if (
+          !organisation ||
+          availableOrganisations.map(orga=>orga.uuid).indexOf(organisation.uuid) === -1
+        ) {
+          const selectedOrganisation = availableOrganisations[0];
+          dispatch(selectOrganisation(selectedOrganisation));
+          dispatch(
+            addNotification(
+              `Organisation "${(selectedOrganisation && selectedOrganisation.name) || 'none' }" selected`,
+              2000
+            )
+          );
+          localStorage.setItem(
+            "lizard-management-current-organisation",
+            JSON.stringify(selectedOrganisation)
+          );
```

Comment:
Why is this here? If we do it, it should happen each time selectedOrganisation is set, and that is done in the selectOrganisation action.

Reply:
Same holds for the notification, I think.

Reply:
Good catch !
I think this was indeed done double.
I movet it to the setOrganisation action.
Could you review once more ?